### PR TITLE
Test-feature

### PR DESCRIPTION
**Pull request by Addis-Belete**
This branch contains:
- tests for logic and
- Snapshot tests for components

### CODE REVIEW

File: src/__tests__/logic/calculate-test.js [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+    expect(result.next).toEqual(null);
+    expect(result.operation).toEqual(null);
+  });
+  it('should return a result of multiplication', () => {
+    const result = calculate('=', { total: '2', next: '0', operation: 'X' });
+    expect(result.total * 1).toEqual(0);
+    expect(result.next).toEqual(null);
+    expect(result.operation).toEqual(null);
+  });
+});
```
Comment (Elbie-em) [inline src/__tests__/logic/calculate-test.js:68]: - [x] Kindly add negative test cases for your unit tests above.

**Review by Elbie-em — CHANGES_REQUESTED** [review]
## STATUS: CHANGES REQUIRED 🟡  ♻️ 

Hi @Addis0943, you have really done an awesome job with your tests. Congrats. However, just one more thing to be done before we move forward. Check my comments below ⬇️ 

*All the best and keep up with the good work*

#### *Request another review once you are done*

> Elbie Moonga (Part-time TSE)

> Github (@Elbie-em)
> Slack (Elbie Moonga)
> Twitter ([@ElbieEm](https://twitter.com/ElbieEm))
> LinkedIn([Elbie Moonga](https://www.linkedin.com/in/elbiemoonga/))

**Comment by Addis-Belete** [comment]
Hello TSE
Thank you for your feedback. I added negative tests. kindly review it.
Regards

**Review by george-shammar — APPROVED** [review]
STATUS :APPROVED :heavy_check_mark: :heavy_check_mark: 

@Addis0943 , Thank you for adding negative tests to your test suites.

Great work :+1: 

I'm happy to approve this CRR.

I wish you good luck with your future projects.

**Cheers** :smile:

**Pull request merged**